### DISCUSSION TITLE
Need help looking into error with -version flag for dotnet watch implementation

### DISCUSSION THREAD

**Discussion by creative-cloud** [cake-build/cake#3273] [Q&A]
#2570
I have been working on a cake alias for dotnet watch and here is my branch - https://github.com/creative-cloud/cake/tree/feature/watch
Flags like -list, -project, -help all seem to work just fine, but I am seeing an issue with the -version flag. **After the version output** I am seeing a super generic error with the dotnet cli-

![image](https://user-images.githubusercontent.com/25395525/110880253-552e1000-82ac-11eb-8590-65dbe0180ba7.png)

I am new to cake and need some guidance on how to debug this. Also can someone run my branch locally as a sanity check, it seems super weird that only one flag would fail.
Thanks!

P.S. @augustoproiete, since you last updated the issue.

**Comment by augustoproiete** [comment]
Hey @creative-cloud, thanks for looking into this.

You might able to see more information about the error if you set the verbosity to `diagnostic` e.g.

```
.\build.ps1 --verbosity=diagnostic
```

I suspect the underlying issue is that `dotnet watch --version` doesn't "do" anything other than to display the version, so your code might be expecting something different. It also does not expect the `--project` argument that you're passing in, so that might be one issue as well.

---

ps: `dotnet watch` is kind of a complex command, so you might want to continue the discussion on https://github.com/cake-build/cake/issues/2570 to agree on the design before submitting a PR (just to save your time, really - though feel free to propose a PR if you prefer/think it's easier to explain the design proposed with code - your call).

[I've added a some initial thoughts](https://github.com/cake-build/cake/issues/2570#issuecomment-797193899) on the issue you referenced with things you might want to consider in this implementation.

**Reply by creative-cloud** [reply]
Hey @augustoproiete , thanks for getting back to me.
I read through your thoughts and shared mine on the issue. Let's agree on a general design to follow before opening a PR.

Coming back to the problem at hand, I agree that --version doesn't _do_ anything per say and that is slightly different from the regular watch command, but what is surprising is that --list is very similar to version and that seems to work perfectly fine. My code is essentially a wrapper that sends out a formed command to the dotnet cli, so the fact replacing --list with --version breaks it has me stumped.
Changing the verbosity does not seem to help with the error coming back from the cli call. 

Also, it looks like watch --version does support the --project flag, (and will show an error without it if you are in the wrong directory) so I am ruling that out as a cause for this.

**Reply by augustoproiete** [reply]
Interesting. So my next guess is that this could be an edge-case of the `.csproj` you're running `dotnet watch` on. And that's because `--version` is one of the arguments that Cake itself uses (i.e. it's a "reserved" argument name) which doesn't expect any other arguments.

Perhaps create a new empty Console app and test your `DotNetWatch` alias on that project instead?
